sql/information_schema/information_schema.go

```diff
@@ -800,6 +800,85 @@ func tableConstraintRowIter(ctx *Context, c *Catalog) (RowIter, error) {
 	return RowsToRowIter(rows...), nil
 }
 
+func getColumnNamesFromIndex(idx Index, table Table) []string {
+	var indexCols []string
+	for _, expr := range idx.Expressions() {
+		col := plan.GetColumnFromIndexExpr(expr, table)
+		if col != nil {
+			indexCols = append(indexCols, fmt.Sprintf("`%s`", col.Name))
+		}
+	}
+
+	return indexCols
+}
+
+func keyColumnConstraintRowIter(ctx *Context, c *Catalog) (RowIter, error) {
+	var rows []Row
+	for _, db := range c.AllDatabases() {
+		tableNames, err := db.GetTableNames(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tableName := range tableNames {
+			tbl, _, err := c.Table(ctx, db.Name(), tableName)
+			if err != nil {
+				return nil, err
+			}
+
+			// Get UNIQUEs, PRIMARY KEYs
+			// TODO: This does not correctly distinguish nonnative indexes used with a sql.Table
```

Comment:
Should say:

Doesn't correctly consider primary keys from table implementations that don't implement sql.IndexedTable